pkg/kapp/app/recorded_app.go

```diff
@@ -208,10 +214,13 @@ func (a *RecordedApp) createOrUpdate(c *corev1.ConfigMap, labels map[string]stri
 
 			return nil
 		}
-
 		return fmt.Errorf("Creating app: %s", err)
 	}
 
+	if isDiffRun {
```

Comment:
I think we can remove the condition. It doesn't matter if it is `diffRun` or not, you should call the setMeta fn

Reply:
Yeah if calll setMeta for all we have bit modify our [`meta()` method](https://github.com/vmware-tanzu/carvel-kapp/blob/6f5f582e71bc41e8c5fb53af5f1c6a4cb57d8d02/pkg/kapp/app/recorded_app.go#L412) to first check for migrated app and then check for memoizedMeta is non nil

Reply:
As if we start setting meta here then `a.memoizedMeta` will never be nil and hence even when migration is enabled it `a.isMigrated ` will not get set to `True` and hence will unable to find the app in `Rename()` method as now we have to get configMap with fully qualified name.

Reply:
And if we start setting the `a.isMigrated` flag here based on ` a.isMigrationEnabled()` then it may be a reduplication of logic

Reply:
@rohitagg2020 we can't start setting meta everytime as in case when migration is enabled (i.e `KAPP_FQ_CONFIGMAP_NAMES=True`) and we set the meta here then `a.isMigrated` flag always remain false as it will always read from cache (i.e. memoizedMeta) and hence won't set the  `a.isMigrated` flag to true and will try to get the configmap with non suffixed name and won't able to find it and give an error.

Reply:
I'm no more setting `isMigrated` anywhere we need to do that when we were calling `setMeta()` for each every call but now as I'm doing it just in case of `--diff-run`  so not needed.

Reply:
Any specific reason for not setting meta in all scenarios?

Reply:
Setting meta in for all scenarios is making an issue in case there is existing migrated app and we try to update it making `KAPP_FQ_CONFIGMAP_NAMES` env variable False. (basically in [this](https://github.com/vmware-tanzu/carvel-kapp/blob/d667147b688102aea0f729741c2841c35e1015d2/test/e2e/create_update_delete_test.go#L194) test scenario)

Reply:
The only concern I have is that we will have to set `isMigrated` to true in multiple scenarios.

Reply:
Yeah we have to set it in both  `CreateOrUpdate()` and `RenamePrevApp()` methods.